Foundation/Collection/Foldable.hs

```diff
@@ -43,6 +45,20 @@ class Foldable collection where
     foldr' :: (Element collection -> a -> a) -> a -> collection -> a
     foldr' f z0 xs = foldl f' id xs z0 where f' k x z = k $! f x z
 
+-- | Fold1's. Like folds, but they assume to operate on a NonEmpty collection.
+class Foldable f => Fold1able f where
```

Comment:
Some types aren't effectively `Fold1able`, like trees or the like because they have no canonical way of unconsing an element.